airbyte-cdk/python/unit_tests/sources/test_connector_state_manager.py

```diff
@@ -3,10 +3,274 @@
+                {"type": AirbyteStateType.STREAM, "stream": {"stream_descriptor": {"name": "accounts"}, "stream_state": {"id": "abc"}}},
+            ],
+            "users",
+            None,
+            {},
+            id="test_get_stream_without_stream_state",
+        ),
+        pytest.param(
+            [{"type": AirbyteStateType.LEGACY, "data": {"users": {"created_at": 12345}, "accounts": {"id": "abc"}}}],
+            "users",
+            "public",
+            {"created_at": 12345},
+            id="test_get_stream_from_legacy_state",
+        ),
+        pytest.param(
+            [
+                {
+                    "type": AirbyteStateType.STREAM,
+                    "stream": {"stream_descriptor": {"name": "users", "namespace": "public"}, "stream_state": {"created_at": 12345}},
+                },
+                {
+                    "type": AirbyteStateType.STREAM,
+                    "stream": {"stream_descriptor": {"name": "accounts", "namespace": "public"}, "stream_state": {"id": "abc"}},
+                },
+            ],
+            "missing",
+            "public",
+            {},
+            id="test_get_missing_stream",
+        ),
+        pytest.param(
+            [
+                {
+                    "type": AirbyteStateType.STREAM,
+                    "stream": {"stream_descriptor": {"name": "users", "namespace": "public"}, "stream_state": {"created_at": 12345}},
+                },
+                {
+                    "type": AirbyteStateType.STREAM,
+                    "stream": {"stream_descriptor": {"name": "accounts", "namespace": "public"}, "stream_state": {"id": "abc"}},
+                },
+            ],
+            "users",
+            "wrong_namespace",
+            {},
+            id="test_get_stream_wrong_namespace",
+        ),
+        pytest.param([], "users", "public", {}, id="test_get_empty_stream_state_defaults_to_empty_dictionary"),
+        pytest.param(
+            [
+                {
+                    "type": AirbyteStateType.STREAM,
+                    "stream": {"stream_descriptor": {"name": "users", "namespace": "public"}, "stream_state": None},
+                },
+            ],
+            "users",
+            "public",
+            {},
+            id="test_get_stream_with_stream_state_none_returns_empty_map",
+        ),
+    ],
+)
+def test_get_stream_state(input_state, stream_name, namespace, expected_state):
+    stream_to_instance_map = {"users": StreamWithNamespace()}
+    state_messages = [AirbyteStateMessage.parse_obj(state_obj) for state_obj in list(input_state)]
+    state_manager = ConnectorStateManager(stream_to_instance_map, state_messages)
+
+    actual_state = state_manager.get_stream_state(stream_name, namespace)
+
+    assert actual_state == expected_state
+    # todo add a test where the state is None and we still return an empty mapping
```

Comment:
Can you do this in this PR?

Reply:
ah yeah i actually had already added it, but forgot to remove the comment. it's the last test in the list of test cases